app/controllers/users_controller.rb

```diff
@@ -277,6 +268,19 @@ def sanitize_user_params
     params[:user].delete_if { |_k, v| v.blank? }
   end
 
+  def default_suggested_users
+    @default_suggested_users ||= User.where(username: DEFAULT_FOLLOW_SUGGESTIONS)
+  end
+
+  def determine_follow_suggestions(current_user)
+    recent_suggestions = Suggester::Users::Recent.new(
+      current_user,
+      attributes_to_select: INDEX_ATTRIBUTES_FOR_SERIALIZATION,
+    ).suggest
+
+    recent_suggestions.empty? ? default_suggested_users : recent_suggestions
```

Comment:
This seems to be the only place in the controller where you use this, so maybe we can move it into `Suggester::Users::Recent` which performs the check itself. Then this method can follow the "tell don't ask" principle :

```ruby
def determine_follow_suggestions(current_user)
  Suggester::Users::Recent.new(
    current_user,
    attributes_to_select: INDEX_ATTRIBUTES_FOR_SERIALIZATION,
  ).suggest
end
```

I'd even wrap up the `new` and `suggest` calls into a combined `call` class method as we do for several other service objects:

```ruby
def self.call(user, attributes_to_select:)
  new(...).suggest
end
```

then this whole method essentially is just

```ruby
Suggester::Users::Recent.call(current_user,  attributes_to_select: INDEX_ATTRIBUTES_FOR_SERIALIZATION)
```

Reply:
I think when I initially PRed this, I was using `default_suggested_users` in two locations (but I have since [removed it](https://github.com/thepracticaldev/dev.to/pull/7637/commits/61aa0a80c9b514228dfd489bc56b87fbda0c838e) when I discovered that it was covering a situation that probably shouldn't ever happen).

I expect to come back to this code next week during cool down, so if it's okay with you, I'm going to hold off on refactoring this and come back to this refactor in a separate PR :) I do think there is definitely room to fix this up, though, so I definitely agree with you there 👍